Home Work № 7

skyseedim opened the pull request.

skyseedim committed:
Home Work № 7

NicknameSoHard commented on HW7.py:
```diff
@@ -0,0 +1,190 @@
+        self.matr = rows
+
+    def __str__(self):  # представление матрицы в виде набора строк
+        self.row = ''
+        for i in range(len(self.matr)):
+            self.par = len(self.matr[i])
+            for j in range(self.par - 1):
+                self.row += str(self.matr[i][j]) + ' '
+            self.row += str(self.matr[i][self.par - 1]) + '\n'
+        return self.row
```
По логике все выглядит хорошо, только я не совсем понимаю для чего `self.row` и  `self.par` сохранять в классе. Это же просто временные переменные, которые при каждом вызове переписываются. Чтобы лишнюю память не забивать просто переменными их можно и оставить.

NicknameSoHard commented on HW7.py:
```diff
@@ -0,0 +1,190 @@
+"""
+
+print("\nTask № 2")
+# 1-й способ решения: создаётся только один класс "Одежа", разделение на типы происходит с помощью декоратора decorator_par
+# На самом деле пытался это сделать с помощью сеттера декоратора property непосредственно в классе. Но в итоге пришел к выводу,
+# что c помощью сеттера можно задать значения только из вне класса, а если значения меняются внутри класса, то он их не задаёт?
```
Нет, при установке внутри класса тоже будет вызываться property. Один из вариантов сделать это задание, да.

NicknameSoHard commented on HW7.py:
```diff
@@ -0,0 +1,190 @@
+
+cl1 = Cell(12)
+cl2 = Cell(15)
+cl1+cl2
+cl1*cl2
+cl1-cl2
+cl2-cl1
+cl1/cl2
+cl2/cl1
+cl1.make_order(7)
+cl2.make_order(7)
```
Отпринитили бы хоть чтоб видеть результат)

skyseedim committed:
Изменение в решение задачи 1

Переписан код для представления матрицы в виде набора строк. Теперь в каждую ячейку вмещается до 4 знаков без смещения  (введён отступ на 4 знака).
